inc/class-core-sitemaps-provider.php

```diff
@@ -162,13 +172,13 @@ public function get_sitemaps() {
 	 *
 	 * By default this is the sub_type as specified in the class property.
 	 *
-	 * @return array List of object types.
+	 * @return array List: containing object types or false if there are no subtypes.
 	 */
 	public function get_object_sub_types() {
-		// FIXME: fix this hack.
-		$c       = new stdClass();
-		$c->name = $this->sub_type;
+		if ( ! empty( $this->sub_type ) ) {
+			return array( $this->sub_type );
+		}
 
-		return array( $c );
+		return array( false );
```

Comment:
This is better. Generally, I'd still prefer returning an empty array when there are no subtypes and make sure all logic using `get_object_sub_types()` includes handling for empty sets as this is a pretty uncommon return value.

Reply:
@joemcgill So I tested the code with the empty array return, and the issue I found is that PHP won't iterate over an empty array so then I had to update logic using this method to first replace the array with something iterable. This makes the implementation of code calling this function in `Core_Sitemaps_provider::get_sitemaps()` more complex. OTOH this  eccentric return value (could be empty string instead if you prefer, but I thought this was clearer) returns an iterable array that checks true against empty() without adding complexity to the caller code. So I considered this a better solution.

Let me know your thoughts, happy to reconsider.

Reply:
👍 thanks for the background. Let's stick with it for now then and adjust if any real issues arise from this.

Reply:
Ok. Clarifying the above comment for future reference with a screenshot.

Reply:
Clarifiation added in the code for future reference: https://github.com/GoogleChromeLabs/wp-sitemaps/pull/72/commits/fc49d9ac7404f4a733bf9d5a35974d4b1f76cc76